src/main/java/teammates/ui/webapi/action/GetSessionResultsAction.java

```diff
@@ -82,25 +91,14 @@ public ActionResult execute() {
 
             try {
                 // TODO optimize the logic layer to get rid of functions that are no longer necessary
-                if (questionId == null) {
-                    if (selectedSection == null) {
-                        bundle = logic.getFeedbackSessionResultsForInstructorWithinRangeFromView(
-                                feedbackSessionName, courseId, instructor.email,
-                                1, Const.FeedbackSessionResults.QUESTION_SORT_TYPE);
-                    } else {
-                        bundle = logic.getFeedbackSessionResultsForInstructorInSection(
-                                feedbackSessionName, courseId, instructor.email, selectedSection,
-                                SectionDetail.EITHER);
-                    }
+                if (questionId == null && selectedSection != null) {
+                    bundle = logic.getFeedbackSessionResultsForInstructorInSection(
+                            feedbackSessionName, courseId, instructor.email, selectedSection, SectionDetail.EITHER);
+                } else if (selectedSection == null && questionId != null) {
+                    bundle = logic.getFeedbackSessionResultsForInstructorFromQuestion(feedbackSessionName, courseId,
+                            instructor.email, questionId);
                 } else {
-                    if (selectedSection == null) {
-                        bundle = logic.getFeedbackSessionResultsForInstructorFromQuestion(feedbackSessionName, courseId,
-                                instructor.email, questionId);
-                    } else {
-                        bundle = logic.getFeedbackSessionResultsForInstructorFromQuestionInSection(
-                                feedbackSessionName, courseId,
-                                instructor.email, questionId, selectedSection, SectionDetail.EITHER);
-                    }
```

Comment:
I think you missed this out during refactoring? i.e. `if (questionId != null && selectedSection != null)`

Reply:
actually there is no such cases, the only cases are either parameter provided

Reply:
> actually there is no such cases, the only cases are either parameter provided

This is wrong.